aten/src/ATen/NamedTensorUtils.cpp

```diff
@@ -174,6 +160,42 @@ static std::bitset<64> compute_included_idxs(IntArrayRef excluded_idxs) {
   return included_idxs;
 }
 
+static void assert_names_equal(DimnameList a, DimnameList b) {
+  TORCH_CHECK(a == b,
```

Comment:
What does this `==` do if the `out` tensor is named with wildcard names?

Reply:
or does that not matter because you propagate before you call this assertion?

Reply:
the `==` checks that the names are literally the same. It does not do a "matching check".

So, for example, the following are disallowed:
- `torch.abs(Tensor[N, C], out=Tensor[N, None])`
- `torch.abs(Tensor[N, None], out=Tensor[N, C])`

Reply:
Do we plan to add support for more combinations later or does outputting to a partially named tensor not make sense?

Reply:
> Do we plan to add support for more combinations later or  does outputting to a partially named tensor not make sense?

It's possible to modify how all of this works by changing this line of the code. I am open to changing the semantics if it makes sense, but right now I am not sure if it makes sense so I put in a pretty conservative check.

Reply:
makes sense. I guess I was expecting it to follow the same kind of matching rules as other ops, but can always be relaxed later.

Reply:
Our semantics for output tensors are a little weird in general. `out=` tensors get automatically resized to what they need to be.

One concrete example is that you can't resize output tensors that are also input tensors:
```
import torch
x = torch.randn(3, 3)
y = torch.randn(3)
y += x
```
If you can't change the size of an input tensor that is also an output tensor, we probably shouldn't allow the user do this for names. The current conservative check enforces this, but we could relax it a little and still have it enforce this.